hmac.cc

```diff
@@ -61,17 +61,23 @@ map<string, vector<TestData> > kTestVectors = {
    {{"0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b0b", "4869205468657265",
      "b0344c61d8db38535ca8afceaf0bf12b881dc200c9833da726e9376c2e32cff7"},
     {"4a656665", "7768617420646f2079612077616e7420666f72206e6f7468696e673f",
-     "5bdcc146bf60754e6a042426089575c75a003f089d2739839dec58b964ec3843"}, }},
+     "5bdcc146bf60754e6a042426089575c75a003f089d2739839dec58b964ec3843"},
+    {"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", "54657374205573696e67204c6172676572205468616e20426c6f636b2d53697a65204b6579202d2048617368204b6579204669727374",
+     "60e431591ee0b67f0d8a26aacbf5b77f8e0bc6213728c5140546040f0ee37f54"}, }},
```

Comment:
Thanks, these seem like useful extra test cases to include.

However, they don't really belong in this table &ndash; they're not test vectors from the specific RFCs mentioned.  Could you pull them out into a separate `HmacKeyTooShort` test case please?

Reply:
They are actually from RFC 4231 - test case 6

Reply:
Ah, sorry, I should have gone back and checked.

In that case, maybe add a `bool` field to `TestData` to indicate whether a `CKR_KEY_SIZE_RANGE` error is allowed?  I think it makes sense to fail the test when that error shouldn't happen.

Reply:
That makes sense. I will add a commit.